Assets/HoloToolkit/Utilities/Scripts/Singleton.cs

```diff
@@ -26,20 +27,29 @@ public static T Instance
         {
             get
             {
-                if (!IsInitialized && searchForInstance)
+                if (IsInitialized)
                 {
-                    searchForInstance = false;
-                    T[] objects = FindObjectsOfType<T>();
-                    if (objects.Length == 1)
-                    {
-                        instance = objects[0];
-                        instance.gameObject.GetParentRoot().DontDestroyOnLoad();
-                    }
-                    else if (objects.Length > 1)
-                    {
-                        Debug.LogErrorFormat("Expected exactly 1 {0} but found {1}.", typeof(T).Name, objects.Length);
-                    }
+                    return instance;
+                }
+
+                if (!searchForInstance)
+                {
+                    return null;
+                }
+
+                // likly first time
+                T[] objects = FindObjectsOfType<T>();
```

Comment:
If we use [`Resources.FindObjectsOfTypeAll<T>();`](https://docs.unity3d.com/ScriptReference/Resources.FindObjectsOfTypeAll.html) we can find disabled GameObject instances as well.

> Contrary to Object.FindObjectsOfType this function will also list disabled objects